File: source/core/Addons/AdminSite/6/setAjaxVisitPropertyClass.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.AdminSite
    '
    Public Class setAjaxVisitPropertyClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core

                Dim ArgList As String = cpCore.docProperties.getText("args")
                Dim Args As String() = Split(ArgList, "&")
                Dim gd As GoogleDataType = New GoogleDataType
                gd.IsEmpty = True
                For Ptr = 0 To UBound(Args)
                    Dim ArgNameValue As String() = Split(Args(Ptr), "=")
                    Dim PropertyName As String = ArgNameValue(0)
                    Dim PropertyValue As String = ""
                    If UBound(ArgNameValue) > 0 Then
                        PropertyValue = ArgNameValue(1)
                    End If
                    Call cpCore.visitProperty.setProperty(PropertyName, PropertyValue)
                Next
                result = remoteQueryController.main_FormatRemoteQueryOutput(cpCore, gd, RemoteFormatEnum.RemoteFormatJsonNameValue)
                result = cpCore.html.main_encodeHTML(result)
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/Processor/Addons/AdminSite/bak/setAjaxVisitPropertyClass.cs


using Contensive.BaseClasses;
using Contensive.Core;
using Models.Entity;
using Controllers;

// 

namespace Contensive.Addons.AdminSite {
    
    // 
    public class setAjaxVisitPropertyClass : Contensive.BaseClasses.AddonBaseClass {
        
        // 
        // ====================================================================================================
        // '' <summary>
        // '' getFieldEditorPreference remote method
        // '' </summary>
        // '' <param name="cp"></param>
        // '' <returns></returns>
        public override object execute(Contensive.BaseClasses.CPBaseClass cp) {
            string result = "";
            try {
                CPClass processor;
                cp;
                CPClass;
                coreClass cpCore = processor.core;
                string ArgList = cpCore.docProperties.getText("args");
                string[] Args = ArgList.Split("&");
                GoogleDataType gd = new GoogleDataType();
                gd.IsEmpty = true;
                for (Ptr = 0; (Ptr <= UBound(Args)); Ptr++) {
                    string[] ArgNameValue = Args[Ptr].Split("=");
                    string PropertyName = ArgNameValue[0];
                    string PropertyValue = "";
                    if ((UBound(ArgNameValue) > 0)) {
                        PropertyValue = ArgNameValue[1];
                    }
                    
                    cpCore.visitProperty.setProperty(PropertyName, PropertyValue);
                }
                
                result = remoteQueryController.main_FormatRemoteQueryOutput(cpCore, gd, RemoteFormatEnum.RemoteFormatJsonNameValue);
                result = cpCore.html.main_encodeHTML(result);
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            
            return result;
        }
    }
}